queries/lua/highlights.scm

```diff
@@ -129,7 +129,13 @@
 (identifier) @variable
 
 ((identifier) @variable.builtin
- (#any-of? @variable.builtin "self" "string" "table" "vim"))
+ (#any-of? @variable.builtin "_G" "_VERSION" "io" "math" "os" "package" "self" "string" "table" "utf8"))
+
+((identifier) @debug
+  (#eq? @debug "debug"))
```

Comment:
Are we even using `@debug` anywhere?
It might be worth delegating this to a separate PR for more languages.

Reply:
godot uses it, that's about it :laughing:

I don't think checking for debug identifiers to set as a debug capture works for every language, and a lot don't support "debugging" built in to the language

Reply:
It is not clear what `debug` is there for. It is just a relic because there is also a Vim highlight group with that name (where you would also not know what it is for)

Reply:
Maybe we should remove it then..?

Reply:
I would really like this getting removed. It's one of the documented built-ins of Neovim though. It was added in as a group of 6 captures that were added at the same time for Neovim and nvim-treesitter with only getting slowly adopeted by nvim-treesitter. Usually, you would only add a new captures they get introduced for a few languages were we see the need for it.

Reply:
Ok, I'll remove it from here and submit another PR removing it as it's pretty useless and hard to understand its purpose.

Reply:
If it can be consistently added to multiple languages it's ok. It would just be good to not of captures that are only used by one or two languages and aren't 100% where they would apply to new languages. Having a capture already in some languages helps to get some context how to use something and to evaluate whether it is actually needed.

Reply:
I don't really see this being able to be added to most languages, just a few exceptions like godot and maybe debug here..

Reply:
Even in vim in the runtime/syntax dir:
```sh
rg -i "hi" | rg "Debug" | tr \: \\n | rg ".vim" | uniq | sort
```
gives a handful of languages (vim has support for 683)
```
8th.vim
aptconf.vim
cl.vim
cmod.vim
d.vim
eiffel.vim
elm.vim
euphoria3.vim
euphoria4.vim
forth.vim
freebasic.vim
fstab.vim
groovy.vim
haskell.vim
help.vim
hollywood.vim
initex.vim
java.vim
javascript.vim
krl.vim
maple.vim
mel.vim
nasm.vim
obse.vim
pike.vim
plaintex.vim
pov.vim
progress.vim
rego.vim
remind.vim
reva.vim
sass.vim
scilab.vim
```

Reply:
Vim's syntax files are a _mess_, with highlight groups added without rhyme or reason (apart from whatever made sense to the maintainer at the time). Let's not propagate old mistakes without thinking.

Reply:
So...we should remove debug then